### ISSUE TITLE
Graph After Training is Different Than Pretrained Model's Graph (MobileNetV2)

### ISSUE THREAD

**Issue by menessahin** [tensorflow/models#8188]
I have trained an image classification model (MobileNetV2) using train_image_classifier.py and the resulting model's graph is different than the pretrained model given at:

https://github.com/tensorflow/models/tree/master/research/slim/nets/mobilenet

My model has extra nodes such as `Logits/Dropout/dropout_1/Cast`. I have asked this as a question at stackoverflow. See below for graph outputs from tensorboard and list of extra nodes :

Pretrained MobileNetV2's graph:
https://i.stack.imgur.com/CozaP.jpg

My model's graph after training:
https://i.stack.imgur.com/ilJbp.jpg

https://stackoverflow.com/questions/60378750/tf-slim-image-classification-model-nodes-are-different-from-given-pretrained-net

But this seems more like a bug or version mismatch to me. How can I make sure, after training I have the same graph structure as the pretrained model (MobileNetV2)?

**Comment by eesweng** [comment]
Also, I have noticed that for each different TF-GPU version, my trained model contains different number of nodes. I have gone back to tensorflow-gpu-1.2, but still there are extra nodes such as fifo_queue or fifo_queue_Dequeue which don't exist in the given pretrained model.

**Comment by marksandler2** [comment]
Training graph is different from inference.  You need to create evaluation graph, see for instance mobilenet.ipynb for an example.
